app/models/policy_host.rb

```diff
@@ -8,10 +8,24 @@ class PolicyHost < ApplicationRecord
   validates :policy, presence: true
   validates :host, presence: true, on: :create
   validates :host_id, presence: true, uniqueness: { scope: :policy }
+  validate :host_supported?
+
+  scope :supported_versions, lambda { |policy_id|
+    profile = Profile.find_by(policy_id: policy_id)
+    profile&.supported_os_versions&.map(&:version)
+  }
 
   def self.import_from_policy(policy_id, host_ids)
     import(host_ids.map do |host_id|
       { host_id: host_id, policy_id: policy_id }
     end)
   end
+
+  def host_supported?
+    if PolicyHost.supported_versions(policy&.id).include?("#{host&.os_major_version}.#{host&.os_minor_version}")
```

Comment:
Your context is already reduced by the major version, so when you construct the query above in a custom way, you can focus only on the minor version IMO.

Reply:
I thought about this but figured to be safe I'd check the major version too because when I was testing, it was letting me create a PolicyHost with a host that wasn't supported based on the `supported_os_versions` of the profile but when I only checked the minor version, it was thinking that it was supported. But maybe I don't need to test this scenario since it's already constrained by the major version anyway?